build.gradle.kts

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+import java.io.ByteArrayOutputStream
+import java.net.URI
+
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+}
+
+plugins {
+    idea
+    `java-library`
+    `maven-publish`
+    signing
+    checkstyle
+    id("com.github.gmazzo.buildconfig") version "3.0.2"
+    id("com.github.spotbugs") version "4.7.2"
+    id("com.diffplug.spotless") version "5.14.2"
+    id("com.github.johnrengelman.shadow") version "6.1.0"
+}
+
+group = "org.mongodb.spark"
+version = "3.2.0-SNAPSHOT"
+description = "The official MongoDB Apache Spark Connect Connector."
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+}
+
+repositories {
+    mavenCentral()
+    maven("https://jitpack.io")
+}
+
+extra.apply {
+    set("mongodbDriverVersion", "[4.3,4.3.99)")
+    set("sparkVersion", "3.1.2")
+
+    // Testing dependencies
+    set("junitJupiterVersion", "5.7.2")
+    set("junitPlatformVersion", "1.7.2")
+
+    // Integration test dependencies
+}
+
+val mongoDependencies: Configuration by configurations.creating
```

Comment:
What's this for?

Reply:
This will be used to build the all jar which users will use with Spark.

Deploying to spark (and Kafka) doesn't use maven do its just add the jar file to the plugins location. So we need to create an uber jar to do that and make it easy.